todo_react/todo-react/src/App.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {useState} from 'react';
+import './App.css';
+import {AddToDo} from "./components/AddToDo";
+import {ToDoItem} from "./components/ToDoItem";
+import {ITodo} from "./models";
+import {TodosContext} from "./context"
+
+export const App: React.FC = () => {
+    let todoList = [...JSON.parse(localStorage.getItem('TodoList') || "")]
+    const [todos, setTodos] = useState<ITodo[]>(todoList)
+    const saveToLocalStorage = (todos: ITodo[]) => {
+        localStorage.setItem('TodoList', JSON.stringify(todos))
+    }
+    const addTodo = (todo: ITodo) => {
+        const newTodos = [...todos, todo]
+        setTodos(newTodos)
+        saveToLocalStorage(newTodos)
+    }
+    const deleteTodo = (todoId: number) => {
+        const newTodos = todos.filter(todoEl => todoId !== todoEl.id)
+        setTodos(newTodos)
+        saveToLocalStorage(newTodos)
+    }
+    const doneTodo = (todoId: number) => {
+        const newTodos = todos.map(todoEl => {
+            if (todoId === todoEl.id) {
+                todoEl.completed = !todoEl.completed
```

Comment:
seems like it's a state modification, please avoid it :)

Reply:
Roman means you shouldn’t change `todoEl` (original object in `todos`). Create new object instead and return it from `.map` callback